src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -997,9 +991,25 @@ protected Iterable<Object> toIterable(ExecutionContext context, ExecutionStrateg
 
     private void handleTypeMismatchProblem(ExecutionContext context, ExecutionStrategyParameters parameters) {
         TypeMismatchError error = new TypeMismatchError(parameters.getPath(), parameters.getExecutionStepInfo().getUnwrappedNonNullType());
-        context.addError(error);
 
-        parameters.getDeferredCallContext().onError(error);
+        addErrorToRightContext(error, parameters, context);
+    }
+
+    // Errors from a deferred field are handled in the deferred execution strategy
+    private static void addErrorToRightContext(GraphQLError error, ExecutionStrategyParameters parameters, ExecutionContext executionContext) {
```

Comment:
I had to change the way we add errors to the `ExecutionContext`. 

Errors resulting from executing deferred fields were being added to the main `ExecutionContext`. This behaviour didn't result in a visible error before because we resolved the full initial response before starting the execution of any deferred field. After the execution order changed, errors from deferred fields were ending up in the initial response.

This PR changes this, so now, errors from deferred fields are only added to the `DeferredCallContext`.

Reply:
We could also add the deferred errors to the `ExecutionContext`, marking them as such, so they would be ignored when building the initial response.
By doing this, one could always check `ExecutionContext` to get a comprehensive list of all the errors that happened during the execution - not sure how useful that would be though. ns.